editor/src/components/canvas/canvas-wrapper-component.tsx

```diff
@@ -73,11 +81,117 @@ export const CanvasWrapperComponent = betterReactMemo(
           />
         ) : null}
         {safeMode ? <SafeModeErrorOverlay /> : <ErrorOverlayComponent />}
+        <ModeSelectButtons />
       </FlexColumn>
     )
   },
 )
 
+interface ModeSelectButtonProps {
+  selected: boolean
+  title: string
+  onMouseDown: () => void
+}
+
+const ModeSelectButton = betterReactMemo('ModeSelectButton', (props: ModeSelectButtonProps) => {
```

Comment:
could you put this in a new file please? the canvas-wrapper-component already does a lot of the mission critical error boundary and logging related tasks, I'd hate to dilute it with new UI code

Reply:
Done